### PR TITLE
docs: add release partners guide

### PR DESCRIPTION
**Pull request by joshblack**
This PR adds in the guide for release partners that our team updates when new releases occur. For this upcoming beta release, this guide details the changes that have been delivered and will point to the appropriate guides for teams to follow instructions on how to try out the releases today.

This guide is missing two links: one for feature flag usage and one for CSS Grid that will be added over the next couple of days by others on the team.

#### Changelog

**New**

- Add a guide for our release partners program

**Changed**

**Removed**

#### Testing / Reviewing

- Read through the May 13th, 2021 update
- Verify that the changes detailed make sense and that the code works
- Make sure that what is shared makes sense for what our team is doing and if we are misrepresenting/misframing any of the work being done
- Verify that there is nothing confusing or misleading with how we phrase changes for this beta release

### CODE REVIEW

File: docs/guides/release-partners.md [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+Next, follow our feature flags guide to enable the appropriate feature flags in
+Sass and React. Here's a quick breakdown of the flags that are available and
+what features they correspond with:
+
+| Feature flag                   | Description                                                                                   |
+| ------------------------------ | --------------------------------------------------------------------------------------------- |
+| `enable-css-custom-properties` | Enable CSS Custom Property support with our design tokens                                     |
+| `enable-css-grid`              | Enable CSS Grid support in Sass and with our React components                                 |
+| `enable-v11-release`           | The default flag for v11 functionality. Enable for all updates not covered in the flags above |
```
Comment (tw15egan) [inline docs/guides/release-partners.md:73]: Have we changed this flag in the code yet?

File: docs/guides/release-partners.md [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+##### Component `size` prop
+
+| Feature flag         | Styles         | React                                |
+| -------------------- | -------------- | ------------------------------------ |
+| `enable-v11-release` | No flag needed | <span aria-label="checked">✅</span> |
+
+The `size` prop in our React codebase was not consistent across components. This
+has been addressed in v11 so that each `size` prop is consistent across the
+codebase. The full set of prop names and sizes is below.
+
```
Comment (tw15egan) [inline docs/guides/release-partners.md:115]:
```suggestion

Note: there will be a codemod to help with automatically updating the size prop in our next beta release

```

File: docs/guides/release-partners.md [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+      - [Spacing token updates](#spacing-token-updates)
+    - [What's coming up](#whats-coming-up)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- prettier-ignore-end -->
+
+## About
+
+This document is for teams participating in our 2021 Release Partner Program for
+the Carbon v11 release. To learn more about this program, or to find out how to
```
Comment (tw15egan) [inline docs/guides/release-partners.md]:
```suggestion
the Carbon v11 release. To learn more about this program or find out how to
```

File: docs/guides/release-partners.md [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+
+## Updates
+
+### May 13th, 2021
+
+This is our first beta release for Carbon v11
+<span aria-label="celebrate">🎉</span>
+
+Our goal with this release was to present release partners with the
+documentation and support needed to preview the functionality that we've
```
Comment (tw15egan) [inline docs/guides/release-partners.md]:
```suggestion
documentation and support needed to preview the functionality we've
```

**Review by tay1orjones — APPROVED** [review]
Looks good to me 👍

File: docs/guides/release-partners.md [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+would love to hear any feedback that you might have in our
+[feedback issue](https://github.com/carbon-design-system/carbon/issues/8657).
+
+##### Design Tokens
+
+| Feature flag         | Styles                               | React |
+| -------------------- | ------------------------------------ | ----- |
+| `enable-v11-release` | <span aria-label="checked">✅</span> | N/A   |
+
+All of the updated design tokens for v11 have been added to each theme and is
```
Comment (tw15egan) [inline docs/guides/release-partners.md]:
```suggestion
All of the updated design tokens for v11 have been added to each theme and are
```

File: docs/guides/release-partners.md [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+
+##### Icon updates
+
+| Feature flag    | Styles | React |
+| --------------- | ------ | ----- |
+| No flags needed | N/A    | N/A   |
+
+Our icons have been updated to optimize for the number of exports from our
+`@carbon/icons-react` package along with simplifying the usage of each icon
+component.
```
Comment (tw15egan) [inline docs/guides/release-partners.md]:
```suggestion
Our icons have been updated to optimize the number of exports from our
`@carbon/icons-react` package and simplify the usage of each icon
component.
```

File: docs/guides/release-partners.md [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+
+In v10, each icon was exported along with its size. For example, for the `add`
+icon we would have the following components: `Add16`, `Add20`, `Add24`, `Add32`.
```
Comment (tw15egan) [inline docs/guides/release-partners.md]:
```suggestion
In v10, each icon was exported along with its size. For example, we would have the following components for the `add` icon: `Add16`, `Add20`, `Add24`, and `Add32`.
```

File: docs/guides/release-partners.md [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+re-exported by `@carbon/react`.
+
+In addition, our team is either tackling or will begin to tackle the following
+efforts over the next couple of sprints:
+
+- Updates to IBM Plex, moving from Google Fonts to a more robust solutution to
+  include more font families and weights, along with the variable font for IBM
+  Plex Sans
+- Inline theming support with our new tokens, including support for zones and
+  the ability to support the layering concepts for our token sy stem
```
Comment (tw15egan) [inline docs/guides/release-partners.md]:
```suggestion
  the ability to support the layering concepts for our token system
```

File: docs/guides/release-partners.md [hunk 9]
```diff
@@ -0,0 +1,256 @@
...
+- The `@carbon/upgrade` CLI tool for running automated codemods
+
+In addition, this release will include the first beta of the `@carbon/styles`
+package which will be our next package for styles. This package will be
+re-exported by `@carbon/react`.
+
+In addition, our team is either tackling or will begin to tackle the following
+efforts over the next couple of sprints:
+
+- Updates to IBM Plex, moving from Google Fonts to a more robust solutution to
```
Comment (tw15egan) [inline docs/guides/release-partners.md]:
```suggestion
- Updates to IBM Plex, moving from Google Fonts to a more robust solution to
```

**Review by tw15egan — APPROVED** [review]
Just a few comments and ran it through Grammarly, feel free to discard any changes that don't make sense. The overall docs look great! 👍🏻 ✅

File: docs/guides/release-partners.md (hunk 1 above)
Comment (joshblack) [reply docs/guides/release-partners.md:73]: @tw15egan not yet! Hoping to get a PR up today 👀

**Pull request merged**